compiler/rustc_ast/src/ast.rs

```diff
@@ -300,9 +300,22 @@ pub enum TraitBoundModifier {
     /// `?Trait`
     Maybe,
 
+    /// `const Trait`
+    Const(Span),
+
     /// `~const Trait`
     MaybeConst(Span),
 
+    /// `const !Trait`
+    //
+    // This parses but will be rejected during AST validation.
+    ConstNegative,
+
+    /// `const ?Trait`
+    //
+    // This parses but will be rejected during AST validation.
+    ConstMaybe,
```

Comment:
Having to add three variants for one new trait bound modifier isn't super great and I'd much prefer throwing out `ast::TraitBoundModifier` in favor of `ast::ty::BoundModifiers` if the latter was just as slim memory-wise as the former.

Reply:
Could you try this in a separate PR? I don't think we need to worry about AST sizes as much, tbh.